internal/resource/url.go

```diff
@@ -725,3 +742,36 @@ func (f *Fetcher) parseARN(arnURL string) (string, string, string, string, error
 	key := strings.Join(urlSplit[1:], "/")
 	return bucket, key, "", regionHint, nil
 }
+
+// FetchConfigDualStack is a function that takes care of fetching Ignition configuration on systems where IPv4 only, IPv6 only or both are available.
+func FetchConfigDualStack(f *Fetcher, userdataURLs map[string]url.URL, fetchConfig func(*Fetcher, url.URL) ([]byte, error)) (types.Config, report.Report, error) {
+	var (
+		data []byte
+		err  error
+	)
+	success := make(chan string, 1)
+
+	fetch := func(ip url.URL) {
+		data, err = fetchConfig(f, ip)
+		if err != nil {
+			f.Logger.Err("fetching configuration for %s: %v", ip.String(), err)
+			return
+		}
+
+		success <- ip.String()
+	}
+
+	if ipv4, ok := userdataURLs[IPv4]; ok {
+		go fetch(ipv4)
+	}
+
+	if ipv6, ok := userdataURLs[IPv6]; ok {
+		go fetch(ipv6)
+	}
+
+	// Wait for one success. (i.e wait for the first configuration to be available)
+	ip := <-success
```

Comment:
Sorry if this is a dumb question, If neither succeed what happens? is there a default timeout?

Reply:
No worries :) I was just making sure I was not missing it. Yeah I think that would work well here.